### PR TITLE
Add etcd package

### PR DESCRIPTION
**Pull request by gpop63**
## What does this PR do?

Adds etcd metricset.
Needs a dashboard.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

### CODE REVIEW

File: packages/etcd/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Compatibility
+
+The etcd `metrics` stream was tested with etcd 3.5.1.
+
+## Metrics
+
+### Metrics
+
+The etcd `metrics` stream collects data from the etcd `metrics` module.
```
Comment (sayden) [inline packages/etcd/_dev/build/docs/README.md:13]: This description is wrong. The package description should be extracted from this file https://github.com/elastic/beats/blob/master/metricbeat/module/etcd/_meta/docs.asciidoc and then each metricset will have its own `[metricset]/_meta/docs.asciidoc` file to extract the info for this file from.

File: packages/etcd/_dev/build/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+It's highly recommended to replace `127.0.0.1` with your server’s IP address and make sure that this page accessible to only you.
```
Comment (sayden) [inline packages/etcd/_dev/build/docs/README.md:15]: Insert the sample event in this doc too. Look at an example of how to do it here: https://raw.githubusercontent.com/elastic/integrations/master/packages/kibana/_dev/build/docs/README.md

Basically you need to use `{{event [metricset]}}` or `{{fields [metricset]}}`

File: packages/etcd/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+title: Etcd
+version: 0.0.1
+license: basic
+description: Collect metrics from etcd
+type: integration
+categories:
+  - custom
+release: ga
+conditions:
+  kibana.version: "^7.15.1"
```
Comment (sayden) [inline packages/etcd/manifest.yml:12]: This should now be `"^7.15.1" || ^8.0.0"`

**Review by sayden — COMMENTED** [review]
Left some comments but it's almost finished

File: packages/etcd/_dev/build/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+## Metrics
+
+When using V2, metrics are collected using Etcd v2 API. When using V3, metrics are retrieved from the /metrics endpoint as intended for Etcd v3.
+
+When using V3, metricsest are bundled into `metrics`. When using V2, metricsets available are `leader`, `self` and `store`.
+
+### metrics
+
+This is the `metrics` endpoint metricset of the etcd module. This metrics is being read from the Etcd V3 endpoint and won’t show any activity regarding Etcd V2.
```
Comment (mtojek) [inline packages/etcd/_dev/build/docs/README.md:18]: > This metrics is being read from the Etcd V3 endpoint and won’t show any activity regarding Etcd V2.

I think this information is redundant as you already described it above, before listing data streams. I would recommend to add 1-2 sentences to every data stream and describe what kind of metrics will the data stream proivde.

File: packages/etcd/_dev/build/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+{{event "metrics"}}
+
+{{fields "metrics"}}
+
+### leader
+
+This is the `leader` metricset of the module etcd. This metrics is being read from the Etcd V2 endpoint and won’t show any activity regarding Etcd V3.
```
Comment (mtojek) [inline packages/etcd/_dev/build/docs/README.md:26]: We don't use "metricsets" anymore. This is a term valid for Beats only. Integrations are using data streams. Please adjust the entire PR accordingly.

File: packages/etcd/data_stream/self/fields/fields.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          description: |
+            amount of time the leader has been leader
+    - name: name
+      type: keyword
+      description: |
+        this member’s name
+    - name: recv
+      type: group
+      fields:
+        - name: appendrequest
```
Comment (mtojek) [inline packages/etcd/data_stream/self/fields/fields.yml:30]: Is it possible to follow the snake case? For example: `append_request`, `bandwidth_rate`

File: packages/etcd/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
+format_version: 1.0.0
+name: etcd
+title: etcd
+version: 0.0.1
+license: basic
+description: Collect metrics from etcd servers with Elastic Agent.
+type: integration
+categories:
+  - datastore
+release: ga
```
Comment (mtojek) [inline packages/etcd/manifest.yml:10]: Let's not release it as "GA", but "experimental" or "beta".

**Review by mtojek — COMMENTED** [review]
Once this PR is merged, do you plan to work on dashboards?

File: packages/etcd/_dev/deploy/variants.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+variants:
+  v1:
+    SERVICE_VERSION: 3.5.1
```
Comment (mtojek) [inline packages/etcd/_dev/deploy/variants.yml:3]: Could you please try also against other versions?

**Comment by gabriel0270** [comment]
> Once this PR is merged, do you plan to work on dashboards?

Yes.

File: packages/etcd/manifest.yml (hunk 7 above)
Comment (gpop63) [reply packages/etcd/manifest.yml]: Yes, I can work on dashboards.

File: packages/etcd/_dev/deploy/variants.yml [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+  v2:
+    SERVICE_VERSION: 3.3.8
+  v3:
```
Comment (mtojek) [inline packages/etcd/_dev/deploy/variants.yml:6]: v1, v2, v3 aren't predefined :)

You can rename them to something more meaningful like v351, v338, v3. Then, please adjust the default one too.

**Review by mtojek — COMMENTED** [review]
I re-reviewed the PR and it looks correct. I left one comment regarding definitions of services for tests.

**Review by mtojek — APPROVED** [review]

**Pull request merged**